apps/shared/account/account.service.ts

```diff
@@ -0,0 +1,81 @@
+import {Injectable} from '@angular/core';
+import {
+  Auth,
+  GithubAuthProvider,
+  GoogleAuthProvider,
+  linkWithPopup,
+  signInWithPopup,
+  unlink,
+  User,
+  UserInfo,
+} from '@angular/fire/auth';
+import {Subject} from 'rxjs';
+
+const DEFAULT_AVATAR_URL = 'https://lh3.googleusercontent.com/a/default-user=s64-c';
+
+@Injectable()
+export class AccountService {
+  loggedInStateChange = new Subject<void>();
+
+  /** Whether a user is logged in currently. */
+  isLoggedIn!: boolean;
+
+  /** The Github account information for the current user, if a user is logged in and has linked their Github account. */
+  githubInfo!: UserInfo | null;
+
+  /** The Google account information for the current user, if a user is logged in. */
+  googleInfo!: UserInfo | null;
+
+  avatarUrl = DEFAULT_AVATAR_URL;
+  displayName: string | undefined;
+
+  constructor(private auth: Auth) {
+    this.auth.onAuthStateChanged((user) => {
+      this.avatarUrl = user?.photoURL || DEFAULT_AVATAR_URL;
```

Comment:
For more explicitness we could use nullish coalescing, same below.

```suggestion
      this.avatarUrl = user?.photoURL ?? DEFAULT_AVATAR_URL;
```

Reply:
We actually don't want nullish coalescing in this case because if the photoURL we get is `''` we want to use the default one instead.